applicationset/services/pull_request/azure_devops.go

```diff
@@ -70,13 +70,21 @@ func (a *AzureDevOpsService) List(ctx context.Context) ([]*PullRequest, error) {
 		SearchCriteria: &git.GitPullRequestSearchCriteria{},
 	}
 
+	pullRequests := []*PullRequest{}
+
 	azurePullRequests, err := client.GetPullRequestsByProject(ctx, args)
 	if err != nil {
+		// A standard Http 404 error is not returned for Azure DevOps,
+		// so checking the error message for a specific pattern. Also please note that the repos are filtered later,
+		// so what is relevant for AzureDevOps is the existence of the project.
```

Comment:
```suggestion
		// so checking the error message for a specific pattern. 
		// NOTE: Since the repos are filtered later, only existence of the project
		// is relevant for AzureDevOps by now
```
nit; Also what's stopping us from filtering the repos beforehand?